runtime/lib/ttnn/program.cpp

```diff
@@ -295,6 +295,57 @@ run(::tt::target::ttnn::ReductionOp const *op, ::ttnn::Device &device,
   }
 }
 
+template <int32_t Rank>
+static std::array<int32_t, Rank>
+vectorToArray(const std::vector<int32_t> &vec) {
+  if (vec.size() != Rank) {
+    throw std::invalid_argument("Vector size does not match array size");
+  }
+  std::array<int32_t, Rank> arr;
+  std::copy(vec.begin(), vec.end(), arr.begin());
+  return arr;
+}
+
+template <int32_t Rank>
+static ::ttnn::Tensor invoke_reshape(const ::ttnn::Tensor &tensor,
+                                     const std::vector<int32_t> &shape) {
+  return ::ttnn::reshape(tensor, vectorToArray<Rank>(shape));
+}
+
+static void
+run(::tt::target::ttnn::ReshapeOp const *op, ::ttnn::Device &device,
+    std::unordered_map<std::uint32_t, ::ttnn::Tensor *> &liveTensors,
+    std::list<::ttnn::Tensor> &tensorPool) {
+  auto &in = *liveTensors.at(op->in()->global_id());
+  const auto *shape_arg_fb_ptr = op->shape();
+  std::vector<int32_t> shape(shape_arg_fb_ptr->begin(),
+                             shape_arg_fb_ptr->end());
+
+  constexpr int32_t Rank1 = 1;
```

Comment:
lowercase `rank` as it's variable name

Reply:
It's template parameter I wanted to leave it upper case seems nicer :)

Reply:
I don't think it's ugly or pretty, I just want to follow some rules because if everyone starts writing "pretty" code by their subjective standards, code will become mess. 
I guess we still don't have a strong ruling on this matter.